activerecord/lib/active_record/encryption/properties.rb

```diff
@@ -53,9 +54,26 @@ def []=(key, value)
         data[key] = value
       end
 
+      # Delegation involves object creation. Defining the commonly called methods
+      # can saves allocations when loading large number of encrypted attributes.
+      def each(*args, &block)
+        data.each(*args, &block)
+      end
+
+      def [](*args)
```

Comment:
This `*args` also allocate. I suspect you'd have similar performance by just adding it to `delegate :==, to: :data` above.

Reply:
No problem. Please re-run your benchmark though, as I need to see if the gain is big enough to justify the change.

Reply:
I've added the benchmark results below.